src/Service.Host/client/src/components/SerialNumber.js

```diff
@@ -0,0 +1,111 @@
+
+    useEffect(() => {
+        if (item !== prevSernosNote) {
+            if (item !== null) {
+                setSernosNote(item);
+                setPrevSernosNote(item);
+            } else {
+                setSernosNote({
+                    sernosNotes: '',
+                    sernosGroup: serialNumber.sernosGroup,
+                    sernosNumber: serialNumber.sernosNumber,
+                    sernosTRef: serialNumber.sernosTRef,
+                    transCode: serialNumber.transCode
+                });
+                setPrevSernosNote(item);
+            }
+        }
+    }, [item, serialNumber, prevSernosNote]);
+
+    const sernosNoteInvalid = () => !sernosNote.sernosNotes;
+
+    const handleFieldChange = (propertyName, newValue) => {
+        setSernosNote({ ...sernosNote, [propertyName]: newValue });
+    };
+
+    const handleSaveClick = () => {
+        const href = getSelfHref(sernosNote);
+        if (href) {
+            updateSernosNote(sernosNote.sernosNoteId, sernosNote);
+        } else {
+            addSernosNote(sernosNote);
+        }
+    };
+
+    return (
+        <TableRow key={serialNumber.sernosTRef}>
+            <TableCell component="th" scope="row">
+                {serialNumber.sernosDate
+                    ? moment(serialNumber.sernosDate).format('YYYY-MM-DD')
+                    : ''}
+            </TableCell>
+            <TableCell align="right">{serialNumber.transCode}</TableCell>
+            <TableCell align="right">{serialNumber.documentNumber}</TableCell>
+            <TableCell align="right">{serialNumber.articleNumber}</TableCell>
+            {editing ? (
+                <Fragment>
+                    <TableCell>
+                        <InputField
+                            label="Notes"
+                            fullWidth
+                            error={sernosNoteInvalid()}
+                            helperText={sernosNoteInvalid() ? 'This field is required' : ''}
+                            value={sernosNote.sernosNotes}
+                            onChange={handleFieldChange}
+                            propertyName="sernosNotes"
+                        />
+                    </TableCell>
+                    <TableCell>
+                        <Button disabled={sernosNoteInvalid()} onClick={handleSaveClick}>
+                            <Done />
+                        </Button>
+                        <Button
+                            onClick={() => {
+                                setEditing(false);
+                                setSernosNote(prevSernosNote);
+                            }}
+                        >
+                            <Clear />
+                        </Button>
+                    </TableCell>
+                </Fragment>
+            ) : (
+                <Fragment>
+                    <TableCell align="right">{sernosNote ? sernosNote.sernosNotes : ''}</TableCell>
```

Comment:
think you can do optional chaining sernosNote?.sernosNotes or even if that doesn't work sernosNote && sernosNote.sernosNotes